### DISCUSSION TITLE
Additional filter for file extraction.

### DISCUSSION THREAD

**Discussion by palichx** [Security-Onion-Solutions/securityonion#3909] [Ideas]
In basic configuration zeek extract file based on mime type. in some case need to add protocol filter(source: HTTP/SMB/SMTP). 
For example i want to extract text/plain. but HTTP traffic contains a lot of text/plain files :).

I understand that i can tune zeek script, but will be good manage it in salt config. 

Thanks!

**Comment by weslambert** [comment]
This can be modified by adjusting the values for the file_extraction pillar shown here:

https://raw.githubusercontent.com/Security-Onion-Solutions/securityonion/master/salt/zeek/fileextraction_defaults.yaml

```
zeek:
  policy:
    file_extraction:
```

**Reply by palichx** [reply]
Hi,
i speak about network protocol filter(not about mime type).
Example: extract "text/plain" from SMB traffic and ignoring this files in HTTP.
working for me:
![image](https://user-images.githubusercontent.com/80513304/115557105-0cf60a80-a2ba-11eb-83da-6b29e399c57d.png)

of course i added "text/plain" into fileextraction_defaults.yaml.
